llvm/test/CodeGen/AIE/aie2/ra/waw_reg_renaming_loop.mir

```diff
@@ -0,0 +1,635 @@
+  ; CHECK-NEXT:   renamable $cm3 = VADD killed renamable $cm0, renamable $cm0, $r25
+  ; CHECK-NEXT:   VST_SRS_D16_S32_ag_idx $p0, $dj0, killed renamable $bmh2, $s0, implicit-def $srsrs_of, implicit $crsat, implicit $crrnd, implicit $crsrssign
+  ; CHECK-NEXT:   VST_SRS_D8_S32_ag_idx $p0, $dj0, killed renamable $cm3, $s0, implicit-def $srsrs_of, implicit $crsat, implicit $crrnd, implicit $crsrssign
+  ; CHECK-NEXT:   PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.2:
+  ; CHECK-NEXT:   liveins: $d0, $dj0, $p0, $r25, $r26, $s0, $x10
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   PseudoRET implicit $lr
+  bb.0.entry:
+  successors: %bb.1
+  liveins: $r0, $p0, $d0, $r25, $r26, $x10, $dj0, $s0
+  LoopStart $r0
+  bb.1:
+    successors: %bb.1, %bb.2
+    liveins: $p0, $d0, $r25, $r26, $x10, $dj0, $s0
+    %1:mxm = VBCST_8  $r25
+    %10:mcmm = VLDA_UPS_S64_S16_ag_idx $s0, $p0, $dj0, implicit-def $srups_of, implicit $crsat
+    %3:mbma = VMUL_F_vmac_bm_core_dense %1, %1, $r25, implicit-def $srfpflags, implicit $crfpmask
+    %4:mbma = VADD_F %3, %3, $r25, implicit-def $srfpflags, implicit $crfpmask
+    %11:mcmm = VADD %10, %10, $r25
+    VST_SRS_D16_S32_ag_idx $p0, $dj0, %4, $s0, implicit-def $srsrs_of, implicit $crsat, implicit $crrnd, implicit $crsrssign
+    VST_SRS_D8_S32_ag_idx $p0, $dj0, %11, $s0, implicit-def $srsrs_of, implicit $crsat, implicit $crrnd, implicit $crsrssign
+    PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  bb.2:
+    liveins: $p0, $d0, $r25, $r26, $x10, $dj0, $s0
+    PseudoRET implicit $lr
+...
+
+# Ignore renaming of general purpose registers.
+---
+name:            gpr_replacement
+alignment:       16
+legalized:       true
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: gpr_replacement
+  ; CHECK: bb.0.entry:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $r0, $r1, $r2, $r8
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   LoopStart $r0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   successors: %bb.1(0x40000000), %bb.2(0x40000000)
+  ; CHECK-NEXT:   liveins: $r1, $r2, $r8
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   renamable $r0 = AND $r1, $r2
+  ; CHECK-NEXT:   renamable $r3 = AND $r1, $r8
+  ; CHECK-NEXT:   renamable $r0 = AND killed renamable $r0, renamable $r3
+  ; CHECK-NEXT:   dead renamable $r0 = AND killed renamable $r3, killed renamable $r0
+  ; CHECK-NEXT:   PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.2:
+  ; CHECK-NEXT:   PseudoRET implicit $lr
+  bb.0.entry:
+  successors: %bb.1
+  liveins: $r0, $r1, $r2, $r8
+  LoopStart $r0
+  bb.1:
+    successors: %bb.1, %bb.2
+    liveins: $r1, $r2, $r8
+    %0:er = AND $r1, $r2
+    %1:er = AND $r1, $r8
+    %2:er = AND %0, %1
+    %3:er = AND %1, %2
+    PseudoLoopEnd <mcsymbol .L_1120>, %bb.1
+  bb.2:
+    PseudoRET implicit $lr
+...
+
+# Ignore vector registers (%130:er), that are not dedicated vector/accumulator
+# registers.
```

Comment:
What does it test extra compared to the test above?

Reply:
a gpr register can have a superclass of a boolean vector register

In a previous implementation of vector identification, I would misidentify gpr register. This test was designed to detect those cases.

Reply:
I don't understand `superclass of a boolean vector register`. At that point of the flow, there are no "types" anymore, we just look at classes. What do you mean by a gpr register that is a "dedicated vector/accumulator register"?

Reply:
In a previous version: 
if i query all the alias Reg classes, and check if any one of them is a vector, I get a `isVector=true` for this case, which is why it is a dedicated unit test. 

Of course with our current vector check implementation, this case does not appear anymore.

Reply:
I think I would remove that test then, as there is no notion of "1 RC maps to 1 type". Rather, a class can accommodate multiple types. But that's important in ISel, here this late in the flow, bringing back types adds more confusion IMO.

Reply:
I removed it for clarity